.grit/patterns/go/exported_loop_pointer.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Fix pointer is shared between loop iterations
+---
+
+`$VALUE` serves as a loop pointer that might be exported from the loop. Since this pointer is shared across loop iterations, the exported reference will consistently point to the last loop value, potentially leading to unintended consequences. To address this issue, duplicate the pointer within the loop to ensure each iteration has its own distinct reference.
+
+### references
+
+- [looppointer](https://github.com/kyoh86/looppointer)
+
+tags: #fix #correctness
+
+```grit
+language go
+
+`for _, $val := range $values { $body }` where {
+    and {
+        $body <: not contains `$val := $val`,
+        $body <: contains `$funcName(&$value)` => `$val := $val \n $funcName(&$value)`,
```

Comment:
This won't work for `intSlice = append(intSlice, &p)` for example.

I think it's actually find to just look for `&$value` whether it's in a function or not.

Reply:
Hi @morgante, updated the pattern for `intSlice = append(intSlice, &p)` and now test are also not failing, after `mac os` update fix.